simulator/resourceapplier/resourceapplier_test.go

```diff
@@ -0,0 +1,521 @@
+				t.Errorf("createPods() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestResourceApplier_createPodsWithFilter(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		podToCreate *corev1.Pod
+		filter      FilteringFunction
+		filtered    bool
+		wantErr     bool
+	}{
+		{
+			name: "create a Pod but it should not be created because of the filter",
+			podToCreate: &corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-1",
+					Namespace: "default",
+					Labels: map[string]string{
+						"ignore": "true",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-1",
+							Image: "image-1",
+						},
+					},
+				},
+			},
+			filter: func(ctx context.Context, resource *unstructured.Unstructured, clients *Clients) (bool, error) {
+				if resource.GetLabels()["ignore"] == "true" {
+					return false, nil
+				}
+				return true, nil
+			},
+			filtered: true,
+			wantErr:  false,
+		},
+		{
+			name: "create a Pod and it should be pass the filter",
+			podToCreate: &corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-1",
+					Namespace: "default",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-1",
+							Image: "image-1",
+						},
+					},
+				},
+			},
+			filter: func(ctx context.Context, resource *unstructured.Unstructured, clients *Clients) (bool, error) {
+				if resource.GetLabels()["ignore"] == "true" {
+					return false, nil
+				}
+				return true, nil
+			},
+			filtered: false,
+			wantErr:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			client, mapper := prepare()
+			options := Options{
+				FilterBeforeCreating: map[schema.GroupVersionResource][]FilteringFunction{
+					{Group: "", Version: "v1", Resource: "pods"}: {tt.filter},
+				},
+			}
+			service := New(client, mapper, options)
+
+			p, err := runtime.DefaultUnstructuredConverter.ToUnstructured(tt.podToCreate)
+			if err != nil {
+				t.Fatalf("failed to convert pod to unstructured: %v", err)
+			}
+			unstructedPod := &unstructured.Unstructured{Object: p}
+			err = service.Create(context.Background(), unstructedPod)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("createPods() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			got, err := client.Resource(corev1.Resource("pods").WithVersion("v1")).Namespace("default").Get(context.Background(), tt.podToCreate.Name, metav1.GetOptions{})
+			if err != nil {
+				if tt.filtered && errors.IsNotFound(err) || tt.wantErr {
+					return
+				}
+				t.Fatalf("failed to get pod when comparing: %v", err)
+			}
```

Comment:
Seems that a branch should be added for `tt.filtered == true` but `err == nil` case.

Reply:
True. And also for `err == nil && tt.wantErr`.